src/IR.cpp

```diff
@@ -246,6 +246,21 @@ Expr Load::make(Type type, const std::string &name, const Expr &index, Buffer<>
     return node;
 }
 
+Expr AddressOf::make(Type type, const std::string &name, const Expr &index, Type elem_type, Buffer<> image, Parameter param) {
```

Comment:
The first argument seems unnecessary, because it's a strict function of the elem_type. Not sure we have a way to get from type_of<int>() to type_of<int *>() in our type system though... Until we do, I'd just set the type of the node to type_of<void *>()